### PR TITLE
Add --version-json to flutter CLI

### LINKED ISSUE #10534: Add --json support to flutter --version
I'd like to capture this data in a tool – without having to parse it.

(The x hours/days ago data, for instance, adds noise when tracking if things have changed between runs)

### PR DESCRIPTION
**Pull request by kevmoo**
closes https://github.com/flutter/flutter/issues/10534

### CODE REVIEW

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart [hunk 1]
```diff
@@ -259,6 +264,12 @@ class FlutterCommandRunner extends CommandRunner<Null> {
       return;
     }
 
+    if (globalResults['version-json']) {
+      flutterUsage.sendCommand('version-json');
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart]: Do we need to have separate analytics for this?  Seems like we should still report the same analytics and just move this into the `if (globalResults['version']) {...}` block

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart [hunk 2]
```diff
@@ -59,6 +60,10 @@ class FlutterCommandRunner extends CommandRunner<Null> {
     argParser.addFlag('version',
         negatable: false,
         help: 'Reports the version of this tool.');
+    argParser.addFlag('version-json',
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart]: I think the user API would be a little cleaner if it was `flutter --version --json`.  You could add a check that enforces that `--json` is only supported if `--version` is specified.

File: packages/flutter_tools/lib/src/version.dart [hunk 3]
```diff
@@ -82,6 +82,15 @@ class FlutterVersion {
     return '$flutterText\n$frameworkText\n$engineText\n$toolsText';
   }
 
+  Map<String, Object> toJson() => <String, Object>{
+        'channel': channel,
+        'repositoryUrl': repositoryUrl ?? 'unknown source',
+        'frameworkRevision': frameworkRevision,
+        'frameworkCommitDate': frameworkCommitDate,
+        'engineRevision': engineRevision,
+        'dartSdkVersion': dartSdkVersion
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/version.dart]: Add a trailing comma

**Comment by kevmoo** [comment]
@tvolkert  PTAL

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart [hunk 4]
```diff
@@ -59,6 +60,9 @@ class FlutterCommandRunner extends CommandRunner<Null> {
     argParser.addFlag('version',
         negatable: false,
         help: 'Reports the version of this tool.');
+    argParser.addFlag('--json',
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart]: s/--//

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart [hunk 5]
```diff
@@ -255,10 +259,21 @@ class FlutterCommandRunner extends CommandRunner<Null> {
...
+      if (globalResults['json']) {
+        status = const JsonEncoder.withIndent('  ').convert(FlutterVersion.instance.toJson());
+      } else {
+        status = FlutterVersion.instance.toString();
+      }
+      printStatus(status);
       return;
     }
 
+    if (globalResults['version-json']) {
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart]: s/version-//

**Comment by kevmoo** [comment]
PTAL

**Review by tvolkert — APPROVED** [review]
LGTM

**Comment by tvolkert** [comment]
@kevmoo do you have any objection to `--json` being renamed `--machine`?  Per #10621 we may add a `--machine` flag to `flutter doctor`, and changing the `--json` flag here would make "machine" consistent across the various commands.

**Comment by kevmoo** [comment]
No objection. Haven't started using it yet – waiting for `alpha` to get bumped.

Thanks for the warning, though. 😄

**Pull request merged**